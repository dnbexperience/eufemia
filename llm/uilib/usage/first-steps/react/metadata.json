{
  "id": "uilib/usage/first-steps/react",
  "name": "React & TypeScript",
  "description": null,
  "group": "usage",
  "slug": "/uilib/usage/first-steps/react/",
  "props": [],
  "events": [],
  "related": [],
  "checksum": "090b7d977ba4be5e2c4c04d199a30a4048416c59f443a56985df2f80629d9c40",
  "source": null,
  "sources": {
    "entry": {
      "local": "src/docs/uilib/usage/first-steps/react.mdx",
      "public": "https://eufemia.dnb.no/uilib/usage/first-steps/react/"
    },
    "props": null,
    "events": null,
    "demos": null
  },
  "version": "10.88.0",
  "generatedAt": "2025-10-29T11:07:22.378Z",
  "schemaVersion": 1
}
