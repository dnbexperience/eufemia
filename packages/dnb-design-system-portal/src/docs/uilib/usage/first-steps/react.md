---
title: 'React'
draft: false
order: 5
---

import ComponentBox from 'Tags/ComponentBox'
import { Button, IconPrimary } from 'dnb-ui-lib/src'
import { hamburger as hamburgerIcon } from 'dnb-ui-lib/src/icons/secondary_icons'

# React JS for the web

The most basic way to use the `dnb-ui-lib` is like this:

## Basic Buttons

<ComponentBox>
{`
<Button text="Basic Button" />
`}
</ComponentBox>

### Importing the Components

To make the [Button](/uilib/components/button) Component work, we have to import it as well:

```js
import { Button } from 'dnb-ui-lib'
```

And here an example with several imports:

```js
import { Button, Icon } from 'dnb-ui-lib/components'
import { H1, P, Link } from 'dnb-ui-lib/elements'
import { hamburger as hamburgerIcon } from 'dnb-ui-lib/icons/secondary_icons'
```

You may have a look at a [Example for React on GitHub](https://github.com/dnbexperience/eufemia/tree/develop/packages/examples/example-react).
