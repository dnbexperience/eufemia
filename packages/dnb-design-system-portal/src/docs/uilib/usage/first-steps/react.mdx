---
title: 'React & TypeScript'
order: 5
---

import {
  BasicButtonsExample,
  EventsExample,
} from 'Docs/uilib/usage/first-steps/Examples'

# React JS for the web

The most basic way to use the `@dnb/eufemia` is like this:

## Basic Buttons

<BasicButtonsExample />

### Importing the Components

To make the [Button](/uilib/components/button) component work, we have to import it as well:

```js
import { Button } from '@dnb/eufemia'
```

And here an example with several imports:

```js
import { Button, Icon } from '@dnb/eufemia/components'
import { H1, P, Link } from '@dnb/eufemia'
import { hamburger as hamburgerIcon } from '@dnb/eufemia/icons'
```

### Events

<EventsExample />

Now, let's jump to the [Button Docs](/uilib/components/button) for more details.

## Examples

Have a look at the [Webpack example app](https://github.com/dnbexperience/eufemia-examples/tree/main/packages/example-react) and [SSR example](https://github.com/dnbexperience/eufemia-examples/tree/main/packages/example-ssr).
