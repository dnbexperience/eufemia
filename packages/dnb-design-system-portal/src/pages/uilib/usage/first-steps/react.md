---
header: 'UI Library'
title: 'React'
draft: false
order: 5
---

import { Button, IconPrimary } from 'dnb-ui-lib/src'
import { ButtonEventExample } from 'Pages/uilib/usage/first-steps/examples/react-examples'

# React JS for the web

The most basic way to use the `dnb-ui-lib` is like this:

### Basic Buttons

```jsx
<Button text="Basic Button" />
```

<div className="example-box">
  <Button text="Basic Button" />
</div>

### Large Buttons & Icons

```jsx
<Button
  variant="secondary"
  text="Secondary Button"
  icon="chevron-right_medium"
  size="large"
/>
<Button
  icon="chevron-right"
  icon_size="medium"
  size="large"
/>
```

<div className="example-box">
  <Button variant="secondary" text="Secondary Button" icon="chevron-right_medium" size="large"  />
  <Button icon="chevron-right" icon_size="medium" size="large" />
</div>

## Extended example

```js
import { Button } from 'dnb-ui-lib'

const MyComponent = props => (
  <Button text="Custom Element" variant="tertiary" {...props} />
)
```

## Event example

```js
import React from 'react'
import { Button, Icon } from 'dnb-ui-lib/src'
import { hamburger_medium as hamburgerIcon } from 'dnb-ui-lib/src/icons/secondary_icons_medium'

const onClickHandler = event => {
  console.log('onClickHandler', event)
}

export const MyComponent = () => {
  return (
    <Button on_click={onClickHandler}>
      <Icon icon={hamburgerIcon} />
    </Button>
  )
}
```

<div className="example-box">
  <ButtonEventExample />
</div>
