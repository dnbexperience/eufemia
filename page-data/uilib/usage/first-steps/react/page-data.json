{"componentChunkName":"component---src-docs-uilib-usage-first-steps-react-mdx","path":"/uilib/usage/first-steps/react/","result":{"pageContext":{"frontmatter":{"title":"React & TypeScript","order":5}}},"staticQueryHashes":["1307807442","2315317908","289552748"],"slicesMap":{}}