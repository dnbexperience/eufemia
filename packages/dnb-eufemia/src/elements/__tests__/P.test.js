/**
 * Element Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  toJson,
  axeComponent,
} from '../../core/jest/jestSetup'
import Component from '../P'
import { render } from '@testing-library/react'

const props = fakeProps(require.resolve('../P'), {
  optional: true,
})
props.size = 'x-small'
props.element = 'p'

describe('P element', () => {
  it('have to match default P element snapshot', () => {
    const Comp = mount(<Component {...props} />)
    expect(toJson(Comp)).toMatchSnapshot()
  })
  it('has correct size when size is defined', () => {
    render(<Component size="large" />)
    const element = document.querySelector('.dnb-p__size--large')

    expect(Array.from(element.classList)).toEqual([
      'dnb-p',
      'dnb-p__size--large',
    ])
  })
  it('has correct style when size and a modifier is defined', () => {
    render(<Component size="medium" modifier="medium" />)
    const element = document.querySelector('.dnb-p__size--medium')

    expect(Array.from(element.classList)).toEqual([
      'dnb-p',
      'dnb-p--medium',
      'dnb-p__size--medium',
    ])
  })
  it('has correct style when several modifiers are defined', () => {
    render(<Component modifier="medium small" />)
    const element = document.querySelector('.dnb-p__size--small')

    expect(Array.from(element.classList)).toEqual([
      'dnb-p',
      'dnb-p--medium',
      'dnb-p__size--small',
    ])
  })
  it('has correct style when medium is set to true', () => {
    render(<Component medium />)
    const element = document.querySelector('.dnb-p--medium')
    expect(Array.from(element.classList)).toEqual([
      'dnb-p',
      'dnb-p--medium',
    ])
  })
  it('has correct style when bold is set to true', () => {
    render(<Component bold />)
    const element = document.querySelector('.dnb-p--bold')

    expect(Array.from(element.classList)).toEqual(['dnb-p', 'dnb-p--bold'])
  })
  it('should validate with ARIA rules as a p element', async () => {
    const Comp = render(<Component {...props} />)
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})
