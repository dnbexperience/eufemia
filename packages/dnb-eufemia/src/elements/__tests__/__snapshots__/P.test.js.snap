// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`P element have to match default P element snapshot 1`] = `
<P
  bold={true}
  className="className"
  element="p"
  medium={true}
  modifier="modifier"
  size="x-small"
  small={true}
>
  <ForwardRef
    as="p"
    className="dnb-p  dnb-p--modifier dnb-p--medium className dnb-p__size--x-small"
  >
    <ElementInstance
      as="p"
      className="dnb-p  dnb-p--modifier dnb-p--medium className dnb-p__size--x-small"
      innerRef={null}
    >
      <p
        className="dnb-p  dnb-p--modifier dnb-p--medium className dnb-p__size--x-small"
      >
        children
      </p>
    </ElementInstance>
  </ForwardRef>
</P>
`;
